File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSFlashImageMessageCellData.m
//
//  MSFlashImageMessageCellData.m
//  BlackFireIM
//
//  Created by benny wang on 2022/1/25.
//

#import "MSFlashImageMessageCellData.h"
#import "MSIMSDK-UIKit.h"


@implementation MSFlashImageMessageCellData

- (instancetype)init
{
    self = [super init];
    if (self) {
    }
    return self;
}

- (CGSize)contentSize
{
    return CGSizeMake(200, 200);
}

- (CGFloat)heightOfWidth:(CGFloat)width
{
    CGFloat height = 0;
    if (self.showName) {
        height += 25;
    }
    CGSize containerSize = [self contentSize];
    height += containerSize.height;
    if (self.direction == MsgDirectionOutgoing) {
        height += 20;
    }
    height += 5 + 5;
    return height;
}

- (NSString *)reuseId
{
    return TFlashImageMessageCell_ReuseId;
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSFlashImageMessageCellData.swift
//
//  MSFlashImageMessageCellData.swift
//  MSIM-Swift
//
//  Created by benny wang on 2022/1/26.
//

import UIKit
import MSIMSDK


open class MSFlashImageMessageCellData: MSMessageCellData {

    public var flashElem: MSIMFlashElem {
        return self.elem! as! MSIMFlashElem
    }
    
    public override func contentSize() -> CGSize {
        
        return CGSize(width: 200, height: 200)
    }
    
    public override func heightOfWidth(width: CGFloat) -> CGFloat {
        var height: CGFloat = 0
        if showName {
            height += 20
        }
        let containerSize = contentSize()
        height += containerSize.height
        if direction == .outGoing {
            height += 20
        }
        height += 5 + 5
        return height
    }
    
    public override var reUseId: String {
        return MSMcros.TFlashImageMessageCell_ReuseId
    }
}
